services/ui-src/src/components/fields/ChoiceListField.tsx

```diff
@@ -83,6 +80,28 @@ export const ChoiceListField = ({
     });
   };
 
+  const clearNestedValues = (choices: FieldChoice[]) => {
+    return choices.forEach((choice: FieldChoice) => {
+      // if a choice is not selected and there are children, clear out any saved data
+      if (!choice.checked && choice.children) {
+        choice.children.map((child) => {
+          if (child.type === "radio" || child.type === "checkbox") {
+            form.setValue(child.id, [], { shouldValidate: true });
+            if (child.props?.choices) {
+              child.props.choices.forEach((choice: FieldChoice) => {
+                choice.checked = false;
+              });
+              clearNestedValues(child.props.choices);
+            }
+          } else {
+            form.setValue(child.id, "", { shouldValidate: true });
+          }
+        });
+      }
+      setCheckedOrUnchecked(choice);
```

Comment:
Thought: Since this action isn't really related to the function, it could live outside the function to keep things easier to reason about and ensure that the function has only one purpose. If it needs to be invoked on the nested child fields, though, that would be another thing. I don't think it needs to, though because they will get it run on them when they are rendered in their own ChoiceListField instance.